### ISSUE TITLE
Inline asm doesn't complain about invalid clobbers

### ISSUE THREAD

**Issue by Vlad-Shcherbina** [rust-lang/rust#71362]
### To reproduce

Try compiling the following program:
```rust
#![feature(asm)]
fn main() {
    unsafe {
        asm!("" : : : "whatever");
    }
}
```

### Expected result

It should fail with an error
```
unknown register name ‘whatever’ in ‘asm’
```
Like how gcc and clang do in inline asm in C.

### Actual result

It compiles without an error.

----

Version info:
```
> rustc --version --verbose
rustc 1.44.0-nightly (52fa23add 2020-04-18)
binary: rustc
commit-hash: 52fa23add6fb0776b32cc591ac928618391bdf41
commit-date: 2020-04-18
host: x86_64-pc-windows-msvc
release: 1.44.0-nightly
LLVM version: 9.0
```

**Comment by Amanieu** [comment]
This issue does not apply to the new `asm!` ([RFC 2850](https://github.com/rust-lang/rfcs/pull/2850)) which properly validates all register names.

The legacy `llvm_asm!` is deprecated and is no longer maintained.
